spec/unit/middleware/retry_exceptions_spec.rb

```diff
@@ -160,34 +160,50 @@ def perform_and_track_delays
         end
 
         context 'with an exponential backoff retry strategy' do
-          before do
+          it 'generates an exponential delay' do
             container.instance_eval do
               use_backoff_strategy exponential(10)
             end
-          end
 
-          it 'uses an exponential delay' do
             delays = perform_and_track_delays
+
             expect(delays).to eq([10, 100, 1_000, 10_000, 100_000])
           end
-        end
 
-        context 'with an exponential backoff retry strategy and fuzz factor' do
-          before do
+          it 'generates an exponential delay using explicitly given factor' do
+            container.instance_eval do
+              use_backoff_strategy exponential(10, factor: 3)
+            end
+
+            delays = perform_and_track_delays
+
+            expect(delays).to eq([10, 30, 90, 270, 810])
+          end
+
+          it 'when fuzz_factor given, dissipate delays over range' do
             container.instance_eval do
-              use_backoff_strategy exponential(10, fuzz_factor: 0.5)
+              use_backoff_strategy exponential(10, fuzz_factor: 0.3)
+            end
+
+            delays = perform_and_track_delays
+
+            delays.zip([10, 100, 1_000, 10_000, 100_000]).each do |actual, unfuzzied|
+              expect(actual).not_to eq(unfuzzied)
+              expect(actual).to be_within(30).percent_of(unfuzzied)
             end
```

Comment:
This is prone to producing a false positive because `[].zip([10, 100, 1_000, 10_000, 100_000])` returns `[]` and `[].each` is a no-op -- so if there was some bug or problem in `perform_and_track_delays` such that it returned a blank array, this example would pass when it should not.

If you switch up the zip to `[10, 100, 1_000, 10_000, 100_000].zip(delays)` (and swap the order of the block args) it will fix that problem.